extended-utxo-spec/extended-utxo-specification.tex

```diff
@@ -325,49 +284,45 @@ \subsection{Basic notation}
 %The concatenation of two lists $\lambda_1$ and $\lambda_2$
 %  is denoted $\lambda_1 ::: \lambda_2$.
 
-  \item For two types $K$ and $V$, $\Map{K}{V}$ denotes the type of
-    partial functions from \textit{keys} in $K$ to \textit{values} in
-    $V$, and the result of looking up a key $k$ in a map $m$ is
-    denoted by $m[k]$.  If $m$ is such map then the \textit{domain} of
-    $m$ is $\dom m = \{k \in K: f[k] \mbox{ is defined}\}$; we can
-    regard $m$ as a \textit{total} function from $\dom m$ to $V$, and
-    for expository purposes we will sometimes regard $m$ as a subset
-    of $\dom m \times V$ in the usual way.
-
-    Equality for maps is defined in the obvious way: $m_1 = m_2$ if
-    and only if $\dom m_1 = \dom m_2$ and $m_1[k] = m_2[k]$ for all
-    $k \in \dom m_1$ (assuming we have decidable equality in $V$).
-    
-  \item If the type $M$ is a monoid with binary operation $+$ (for
-    example, a numeric type) the we define the sum of two maps
-    $f, g \in \Map{K}{M}$ to be the map $f+g \in \Map{K}{M}$ with
-    domain $\dom (f+g) = \dom f \cup \dom g$ given by
-    \[
-    (f+g)[k] =
-    \left\{ \begin{array}{ll}
-        f[k] + g[k] & \mbox{if $k \in \dom f \cap \dom g$}\\
-        f[k] & \mbox{if $k \in \dom f \setminus \dom g$}\\
-        g[k] & \mbox{if $k \in \dom g \setminus \dom f$}
-      \end{array}
-      \right.
-      \]
-      We generalise this to the sum of a finite number of maps using
-      the $\sum$ notation in the usual way; care is required if the
-      operation $+$ is non-commutative, since then the sum of a number of
-      maps will depend on the order in which they appear. Note that
-      the type $\Map{K}{M}$ itself becomes a monoid under the
-      operation $+$ which we have just defined, with the empty map $\{\}$ as
-      identity element.
-
-    \item $x \mapsto f(x)$ denotes an anonymous function.
-
-  \item A cryptographic
-    collision-resistant hash of a value $c$ is denoted $c^{\#}$.
+\item $x \mapsto f(x)$ denotes an anonymous function.
+
+\item A cryptographic collision-resistant hash of a value $c$ is denoted $\hash{c}$.
+
+\item For a type $A$ which forms a partial order, $\extended{A}$ is the same
+  type extended with a top element $\infty$ and a bottom element $-\infty$.
 
+\item For a type $A$ which forms a total order,  $\Interval{A}$ is the type
+  of intervals over that type, whose endpoints may be either closed or open.
+  The type $\Interval{A}$ forms a lattice.
 \end{itemize}
-\caption{Basic notation}
-\label{fig:basic-notation}
-\end{ruledfigure}
+
+\subsection {Finitely supported functions}
```

Comment:
The monoidal behaviour is clearer. If you think about maps, you have to worry about e.g. whether `{}` and `{x:0}` are the same. We do want them to be the same, but I think the *reason* for that is that they represent the same function! So since it's a spec, I think we should just use the more abstract notion.

Reply:
aha, that's neat! But, do you crash into the problem again when you come to implement it?

Reply:
Nope. The question changes: instead you just ask "how do I implement a FSF using a map?" The note tells you what to do, and if you keep that in mind the problem shouldn't even arise.